report.c

```diff
@@ -49,19 +48,15 @@ extern int af;
 extern int reportwide;
 
 
-char *get_time_string (void) 
-{
-  time_t now; 
-  char *t;
-  now = time (NULL);
-  t = ctime (&now);
-  t [ strlen (t) -1] = 0; // remove the trailing newline
-  return t;
-}
-
 void report_open(void)
 {
-  printf ("Start: %s\n", get_time_string ());
```

Comment:
I like the old version better here. 
The reason is that you can read "report_open" and understand what it is trying to do just by reading the words in the function. 
Similarly, the function "get_time_string" does just what its name implies. 

So... in two places the "get_time_string" DOCUMENTS what the code is doing, even though it is not efficient and if you're looking at the details, you get an extra indirection.